src/swipes.py

```diff
@@ -0,0 +1,337 @@
+    output_path = '{}{}'.format(EATERY_DATA_PATH, output_file)
+    final_grouping.to_csv(output_path, header=True, index=False)
+    return output_path
+  except Exception as e:
+    print('Failed at sort_by_timeblock')
+    print('Data update failed:', e)
+
+
+def sort_by_day(input_file_path, output_file='daily-averages.csv'):
+  """Sorts and runs average swipe/time calculations.
+  Saves to a csv file or updates files if one already exists
+  Currently not used in main code, but allows for future use of data
+
+  Keyword arguments:
+  input_file_path -- the file path to our input csv file with all swipe logs, string
+  output_file -- the file name for the output csv file (default 'daily-averages.csv')
+
+  Table columns:
+  weekday -- weekday of swipe event, string (ex: monday, tuesday, etc.)
+  in_session -- whether or not school is on a break, boolean
+  location -- eatery of swipe event, string (some locations have different names than dining.now's)
+  swipes -- number of swipes, int
+  counter -- number of events within this timeblock (used to calculate average), int
+  average -- average number of swipes in this timeblock, float (2 decimals)
+  """
+  try:
+    df = pd.read_csv(input_file_path)
+    df = df.drop(columns=['start_time', 'end_time', BRB_ONLY, DINING_HALL, TRILLIUM])
+    # sum together the swipes of rows with same day/eatery makeup
+    swipes_grouped = df.groupby(['date', 'session_type', 'weekday', 'location']).sum().reset_index()
+    # count the number of times a location/date pair occur
+    location_grouped = swipes_grouped.groupby(['weekday', 'location', 'swipes', 'session_type']).count().reset_index()
+    counter_rename = location_grouped.rename(index=str, columns={'date': 'counter'})
+    # sum swipes and counters for rows with the same weekday and location
+    final_grouping = counter_rename.groupby(['weekday', 'location', 'session_type']).sum().reset_index()
+    final_grouping['average'] = np.around(np.divide(final_grouping['swipes'], final_grouping['counter']), decimals=2)
+    final_grouping.sort_values(by=['location', 'weekday'])
+    final_grouping.to_csv('{}{}'.format(EATERY_DATA_PATH, output_file), header=True, index=False)
+  except Exception as e:
+    print('Faled at sort_by_day')
+    print('Data update failed:', e)
+
+
+def export_data(file_path):
+  """Transforms our tabular data into custom objects to be placed in Eatery objects
+
+  Keyword arguments:
+  file_path -- the file path to our input csv file with all wait time data, string
+  """
+  global breaks
+  global weekdays
+
+  try:
+    df = pd.read_csv(file_path)
+    data = {}
+    today = date.today()
+    session_type = sort_session_type(today, breaks)
+    for location in df['location'].unique():
+      eatery_name = LOCATION_NAMES[location]['name']
+      # look at information that pertains to today's criteria
+      today_df = df.loc[(df['location'] == location) & (df['weekday'] == weekdays[today.weekday()])]
```

Comment:
`and` instead of `&` ?

Reply:
actually we need the `&` for pandas df.  just how it works